Assignment/Assignment/PingProcess.cs

```diff
@@ -13,54 +13,87 @@ public record struct PingResult(int ExitCode, string? StdOutput);
+            {
+                Process process = RunProcessInternal(StartInfo, UpdateStdOutput, default, cancellationToken);
+                cancellationToken.ThrowIfCancellationRequested();
+                return new PingResult(process.ExitCode, stringBuilder?.ToString());
+            }, cancellationToken);
+
+        return await task;
     }
 
-    async public Task<PingResult> RunAsync(params string[] hostNameOrAddresses)
+    async public Task<PingResult> RunAsync(IEnumerable<string> hostNameOrAddresses,
+        CancellationToken cancellationToken = default)
     {
-        StringBuilder? stringBuilder = null;
-        ParallelQuery<Task<int>>? all = hostNameOrAddresses.AsParallel().Select(async item =>
+        StringBuilder? stringBuilder = new();
+        ParallelQuery<Task<PingResult>>? all = hostNameOrAddresses.AsParallel().Select(async item =>
         {
-            Task<PingResult> task = null!;
-            // ...
+            Task<PingResult> task = Task.Run(() => Run(item), cancellationToken);
 
             await task.WaitAsync(default(CancellationToken));
-            return task.Result.ExitCode;
+            return task.Result;
         });
 
         await Task.WhenAll(all);
-        int total = all.Aggregate(0, (total, item) => total + item.Result);
+        int total = all.Aggregate(0, (total, item)
+            => total + item.Result.ExitCode); // 0
+        // update stringBuilder: "The order of the items in the stdOutput is irrelevent and expected to be intermingled"
+        // thank you for doing aggregate above as an example/remider
+        stringBuilder.Append(all.Aggregate("", (compiledString, currentString)
+            => compiledString.Trim() + currentString.Result.StdOutput).Trim());
+
+        // int? lineCount = stringBuilder?.ToString().Split(Environment.NewLine).Length; // format stringBuilder to 40 lines
         return new PingResult(total, stringBuilder?.ToString());
     }
 
-    async public Task<PingResult> RunLongRunningAsync(
-        string hostNameOrAddress, CancellationToken cancellationToken = default)
+    //not passing tests idk if it's me or my OS
+    async public Task<PingResult> RunLongRunningAsync(string hostNameOrAddress, CancellationToken cancellationToken = default)
     {
-        Task task = null!;
-        await task;
-        throw new NotImplementedException();
+        StartInfo.Arguments = hostNameOrAddress;
+        StringBuilder? stringBuilder = null;
+        void UpdateStdOutput(string? line) =>
+            (stringBuilder ??= new StringBuilder()).AppendLine(line);
+        
+        Task<PingResult> task = Task.Factory.StartNew(() =>
+        {
+            Process process = RunProcessInternal(StartInfo, UpdateStdOutput, default, cancellationToken); 
+            return new PingResult(process.ExitCode, stringBuilder?.ToString());
+        }, cancellationToken, TaskCreationOptions.LongRunning, TaskScheduler.Current);
+        
+        return await task;

```

Comment:
Just return your task
```suggestion
        return task;

```